crates/bevy_text/src/text2d.rs

```diff
@@ -168,26 +156,28 @@ pub fn update_text2d_layout(
         Entity,
         Changed<Text>,
         &Text,
-        Option<&Text2dBounds>,
-        &mut Text2dSize,
+        &Text2dBounds,
         Option<&mut TextLayoutInfo>,
     )>,
 ) {
     // We need to consume the entire iterator, hence `last`
     let factor_changed = scale_factor_changed.iter().last().is_some();
     let scale_factor = windows.scale_factor(WindowId::primary());
 
-    for (entity, text_changed, text, maybe_bounds, mut calculated_size, text_layout_info) in
-        &mut text_query
-    {
+    for (entity, text_changed, text, bounds, text_layout_info) in &mut text_query {
         if factor_changed || text_changed || queue.remove(&entity) {
-            let text_bounds = match maybe_bounds {
-                Some(bounds) => Vec2::new(
-                    scale_value(bounds.size.x, scale_factor),
-                    scale_value(bounds.size.y, scale_factor),
-                ),
-                None => Vec2::new(f32::MAX, f32::MAX),
-            };
+            let text_bounds = Vec2::new(
+                if bounds.size.x != f32::MAX {
+                    scale_value(bounds.size.x, scale_factor)
+                } else {
+                    f32::MAX
+                },
+                if bounds.size.x != f32::MAX {
+                    scale_value(bounds.size.y, scale_factor)
+                } else {
+                    f32::MAX
+                },
```

Comment:
Could this be written as
```suggestion
                scale_value(bounds.size.x, scale_factor).min(f32:MAX),
                scale_value(bounds.size.y, scale_factor).min(f32::MAX),
```
or would that not be correct / work?

Reply:
It seems like using min should be okay. This part worried me slightly, that's why I wrote it in that clumsy way. 
Perhaps the bounds should be optional instead, but that feels redundant.

Reply:
Changed Text2dBounds to use f32::INFINITY instead of f32::MAX to represent unbounded lengths instead, then there is no need for any comparisons.